src/pkg/repository/repository_unexported_test.go

```diff
@@ -25,22 +39,66 @@ func TestRepositoryModelSuite(t *testing.T) {
 	})
 }
 
-func (suite *RepositoryModelSuite) TestWriteGetModel() {
+func (suite *RepositoryModelIntgSuite) SetupSuite() {
 	ctx, flush := tester.NewContext()
 	defer flush()
 
 	var (
-		t        = suite.T()
-		s        = tester.NewPrefixedS3Storage(t)
-		kopiaRef = kopia.NewConn(s)
+		t   = suite.T()
+		s   = tester.NewPrefixedS3Storage(t)
+		k   = kopia.NewConn(s)
+		err error
 	)
 
-	require.NoError(t, kopiaRef.Initialize(ctx))
-	require.NoError(t, kopiaRef.Connect(ctx))
+	require.NotNil(t, k)
+	require.NoError(t, k.Initialize(ctx))
 
-	defer kopiaRef.Close(ctx)
+	suite.kopiaCloser = func(ctx context.Context) {
+		k.Close(ctx)
+	}
 
-	ms, err := kopia.NewModelStore(kopiaRef)
+	suite.kw, err = kopia.NewWrapper(k)
+	require.NoError(t, err)
+
+	suite.ms, err = kopia.NewModelStore(k)
+	require.NoError(t, err)
+
+	suite.sw = store.NewKopiaStore(suite.ms)
+}
+
+func (suite *RepositoryModelIntgSuite) TearDownSuite() {
+	ctx, flush := tester.NewContext()
+	defer flush()
+
+	if suite.ms != nil {
+		suite.ms.Close(ctx)
+	}
+
+	if suite.kw != nil {
+		suite.kw.Close(ctx)
+	}
+
+	if suite.kopiaCloser != nil {
+		suite.kopiaCloser(ctx)
+	}
+}
+
+func (suite *RepositoryModelIntgSuite) TestWriteGetModel() {
```

Comment:
nit: name is confusing

Reply:
I agree, but plea innocence.  I didn't come up with the name.